lib/hanami/assets.rb

```diff
@@ -9,167 +9,68 @@ module Hanami
-    #   require 'hanami/assets'
-    #
-    #   Hanami::Assets.load!
-    #
-    # @example Load Via Configuration Block
-    #   require 'hanami/assets'
-    #
-    #   Hanami::Assets.configure do
-    #     # ...
-    #   end.load!
-    def self.load!
-      configuration.load!
+    # @since 2.1.0
+    # @api public
+    def subresource_integrity?
+      config.subresource_integrity.any?
     end
 
-    # Global assets sources
-    #
-    # This is designed for third party integration gems with frontend frameworks
-    # like Bootstrap, Ember.js or React.
-    #
-    # Developers can maintain gems that ship static assets for these frameworks
-    # and make them available to Hanami::Assets.
-    #
-    # @return [Hanami::Assets::Config::GlobalSources]
-    #
-    # @since 0.1.0
-    #
-    # @example Ember.js Integration
-    #   # lib/hanami/emberjs.rb (third party gem)
-    #   require 'hanami/assets'
-    #
-    #   Hanami::Assets.sources << '/path/to/emberjs/assets'
-    def self.sources
-      synchronize do
-        @@sources ||= Config::GlobalSources.new # rubocop:disable Style/ClassVars
-      end
+    # @since 2.1.0
+    # @api public
+    def crossorigin?(source_path)
+      config.crossorigin?(source_path)
     end
 
-    # Duplicate the framework and generate modules for the target application
-    #
-    # @param _mod [Module] the Ruby namespace of the application
-    # @param blk [Proc] an optional block to configure the framework
-    #
-    # @return [Module] a copy of Hanami::Assets
-    #
-    # @since 0.1.0
-    #
-    # @see Hanami::Assets#dupe
-    # @see Hanami::Assets::Configuration
-    def self.duplicate(_mod, &blk)
-      dupe.tap do |duplicated|
-        duplicated.configure(&blk) if block_given?
-        duplicates << duplicated
-      end
-    end
+    private
 
-    # Duplicate Hanami::Assets in order to create a new separated instance
-    # of the framework.
-    #
-    # The new instance of the framework will be completely decoupled from the
-    # original. It will inherit the configuration, but all the changes that
-    # happen after the duplication, won't be reflected on the other copies.
-    #
-    # @return [Module] a copy of Hanami::Assets
-    #
-    # @since 0.1.0
-    # @api private
-    def self.dupe
-      dup.tap do |duplicated|
-        duplicated.configuration = configuration.duplicate
-      end
-    end
+    def manifest
+      return @manifest if instance_variable_defined?(:@manifest)
 
-    # Keep track of duplicated frameworks
-    #
-    # @return [Array] a collection of duplicated frameworks
-    #
-    # @since 0.1.0
-    # @api private
-    #
-    # @see Hanami::Assets#duplicate
-    # @see Hanami::Assets#dupe
-    def self.duplicates
-      synchronize do
-        @@duplicates ||= [] # rubocop:disable Style/ClassVars
-      end
+      @manifest =
+        # TODO: Add tests for the File.exist? check
+        if config.manifest_path && File.exist?(config.manifest_path)
```

Comment:
@timriley FYI - In the context of https://github.com/hanami/assets-js/pull/4, I had to make the following change: before starting the Esbuild watch mode, I had to create an empty manifest, otherwise the process crashes.

**We can assume that there is a manifest file present.**

The only exception is the race condition: if the Ruby server boots up faster than the JavaScript process to create the empty manifest.
Let's see if that happens; in that case, we can introduce a retry policy in this area.